File: Python/2956.py
# 2956. Find Common Elements Between Two Arrays
# https://leetcode.com/problems/find-common-elements-between-two-arrays/

class Solution:
  def findIntersectionValues(self, nums1: List[int], nums2: List[int]) -> List[int]:
    ans1 = len(list(filter(lambda n: n in nums2, nums1)))
    ans2 = len(list(filter(lambda n: n in nums1, nums2)))
    return [ans1, ans2]

File: Rust/2956.rs
/**
 * 2956. Find Common Elements Between Two Arrays
 * https://leetcode.com/problems/find-common-elements-between-two-arrays/
 */

impl Solution {
  pub fn find_intersection_values(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<i32> {
    let ans1: i32 = nums1
      .iter()
      .filter(|&n| nums2.contains(n))
      .count() as i32;

    let ans2: i32 = nums2
      .iter()
      .filter(|&n| nums1.contains(n))
      .count() as i32;

    return [ans1, ans2].to_vec();
  }
}
